src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -6303,6 +6298,44 @@ public TextEditingTargetNotebook getNotebook()
       return notebook_;
    }
    
+   private void setRMarkdownBehaviorEnabled(boolean enabled)
+   {
+      // register idle monitor; automatically creates/refreshes previews
+      // of images and LaTeX equations during idle
+      if (bgIdleMonitor_ == null && enabled)
+         bgIdleMonitor_ = new TextEditingTargetIdleMonitor(this, 
+               docUpdateSentinel_);
+      else if (bgIdleMonitor_ != null)
+      {
+         if (enabled)
+            bgIdleMonitor_.beginMonitoring();
+         else
+            bgIdleMonitor_.endMonitoring();
+      }
+      
+      // set up mathjax
+      if (mathjax_ == null && enabled)
+         mathjax_ = new MathJax(docDisplay_, docUpdateSentinel_, prefs_);
+
+      if (enabled)
+      {
+         // auto preview images and equations
+         new ImagePreviewer(docDisplay_, docUpdateSentinel_, prefs_);
```

Comment:
I believe they are, but there's a remote change for weird behavior here if you repeatedly switch the type, so I've added some safeguards in https://github.com/rstudio/rstudio/commit/4ba6ab9ed74e7559e59cde7d449c5e6c90e49f4c. (also refactored a little of the common code between the mathjax and image previewers into a separate class)